Marlin/Marlin_main.cpp

```diff
@@ -5181,54 +5188,52 @@ inline void gcode_G28() {
+          for (int8_t axis = (points_over_4 ? 11 : 9); axis > 0; axis -= (points_over_4 ? 2 : 4)) {
             setup_for_endstop_or_probe_move();
             z_at_pt[0] += probe_pt(
               cos(RADIANS(180 + 30 * axis)) * (0.1 * delta_calibration_radius),
               sin(RADIANS(180 + 30 * axis)) * (0.1 * delta_calibration_radius), true, 1);
             clean_up_after_endstop_or_probe_move();
           }
-          z_at_pt[0] /= (pp_equals_5 ? 7 : probe_points);
+          z_at_pt[0] /= float(pp_5_points ? 7 : probe_points);
         }
-        if (!pp_equals_1) {  // probe the radius
-          float start_circles = (pp_equals_7 ? -1.5 : pp_equals_6 || pp_equals_5 ? -1 : 0),
-                end_circles = -start_circles;
+        if (!pp_height_only) {  // probe the radius
           bool zig_zag = true;
-          for (uint8_t axis = (probe_mode == -2 ? 3 : 1); axis < 13; 
-               axis += (pp_equals_2 ? 4 : pp_equals_3 || pp_equals_5 ? 2 : 1)) {
-            for (float circles = start_circles ; circles <= end_circles; circles++) {
+          for (uint8_t axis = (probe_mode == -2 ? 3 : 1); axis < 13;
+                       axis += (pp_center_and_towers ? 4 : pp_all_positions ? 2 : 1)) {
+            float offset_circles = (pp_7_points ? (zig_zag ? 1.5 : 1.0) :
+                                    pp_6_points ? (zig_zag ? 1.0 : 0.5) :
+                                    pp_5_points ? (zig_zag ? 0.5 : 0.0) : 0);
+            for (float circles = -offset_circles ; circles <= offset_circles; circles++) {
               setup_for_endstop_or_probe_move();
               z_at_pt[axis] += probe_pt(
-                cos(RADIANS(180 + 30 * axis)) * 
-                (1 + circles * 0.1 * (zig_zag ? 1 : -1)) * delta_calibration_radius, 
-                sin(RADIANS(180 + 30 * axis)) * 
-                (1 + circles * 0.1 * (zig_zag ? 1 : -1)) * delta_calibration_radius, true, 1);
+                cos(RADIANS(180 + 30 * axis)) * delta_calibration_radius *
+                (1 + circles * 0.1 * (zig_zag ? 1 : -1)),
+                sin(RADIANS(180 + 30 * axis)) * delta_calibration_radius *
+                (1 + circles * 0.1 * (zig_zag ? 1 : -1)), true, 1);
               clean_up_after_endstop_or_probe_move();
             }
-            start_circles += (pp_greather_5 ? (zig_zag ? 0.5 : -0.5) : 0);
-            end_circles = -start_circles;
             zig_zag = !zig_zag;
-            z_at_pt[axis] /= (pp_equals_7 ? (zig_zag ? 4.0 : 3.0) :
-                              pp_equals_6 ? (zig_zag ? 3.0 : 2.0) : pp_equals_5 ? 3 : 1);
+            z_at_pt[axis] /= (2 * offset_circles + 1);
           }
         }
-        if (pp_greather_3 && !pp_equals_5) // average intermediates to tower and opposites
-          for (uint8_t axis = 1; axis < 13; axis += 2)
+        if (points_over_3) // average intermediates to tower and opposites
+          for (uint8_t axis = 1; axis <= 11; axis += 2)
```

Comment:
I see what you mean, but for future maintenance I would prefer you left '<13' or change all to '<=12' that magic number actually means do a full circle no matter what the step is by which axis is increased.